src/ComponentFactory.php

```diff
@@ -235,29 +234,31 @@ public function sanitizeFileName( $fileName ) {
 	protected function mapComponentDescriptionToClassName( DOMElement $description ) {
 
 		if ( $description->hasAttribute( 'type' ) ) {
-
 			$className = $description->getAttribute( 'type' );
-
 			$parent = $description->parentNode;
 
-			if (  $parent instanceof DOMElement && $parent->hasAttribute( 'type' ) ) {
+			if ( class_exists( $className ) ) {
```

Comment:
I really like the idea, but putting the check here may lead to the wrong class being used, e.g. for `Html`.  There is a component `Html`, but also a class with that name in the root namespace.
I would propose first trying for `NAMESPACE_HIERARCHY\parent\component`, then `NAMESPACE_HIERARCHY\component`, and only as a last resort check if the class exists in the root namespace directly.